shop.lua

```diff
@@ -5,6 +5,30 @@
 
 local exchange_shop = {}
 
+-- Tool wear aware replacement for contains_item.
+local function list_contains_item(inv, list, stack)
+	for i, list_stack in pairs(inv:get_list(list)) do
+		if list_stack:get_name()  == stack:get_name()  and
+		   list_stack:get_count() >= stack:get_count() and
+		   list_stack:get_wear()  <= stack:get_wear() then
+			return true, i
```

Comment:
No need for two return values. Once a stack is found, an `if` check would accept it because it is non-nil and non-false.
```lua
if 5 then
	print("foobar")
end

--> Outputs "foobar"
```
Otherwise excellent formatted code!

EDIT: Also store the stack name in a local variable to speed up the execution time. This function is quite often called for each item field in "Owner gives" and "Owner wants"